File: java8/src/java8/Main06.java
package java8;

import java.util.stream.IntStream;

/** Streams. */
public class Main06 {

  public static void main(String[] args) {
    IntStream list = IntStream.range(1, 1000);
    int total = list.parallel().filter(e -> e > 1).reduce(0, (a, b) -> a + b);
    System.out.println(total);
  }

}

File: java8-scala/src/java8/Main06.scala
package java8

import scala.collection.mutable.Buffer

/** An example of custom lambdas. */
object Main06 {
  def main(args: Array[String]) {
    val r = 1 until 1000
    val total = r.par.filter { _ > 1 }.reduce { _ + _ }
    println(total)
  }

}